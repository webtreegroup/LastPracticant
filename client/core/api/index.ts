export * from './api';
export * from './api.consts';
export * from './auth.api';
export * from './base.api';
export * from './forum.api';
export * from './leaderboard.api';
export * from './oauth.api';
export * from './profile.api';
export * from './settings.api';
