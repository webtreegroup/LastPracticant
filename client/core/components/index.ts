export * from './ErrorBoundary';
export * from './GameCanvas';
export * from './PageLayout';
export * from './Logo';
export * from './Meta';
export * from './OAuth';
export * from './GameSoundTheme';
