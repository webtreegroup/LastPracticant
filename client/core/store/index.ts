export * from './store';
export * from './store.types';
export * from './actions';
export * from './selectors';
