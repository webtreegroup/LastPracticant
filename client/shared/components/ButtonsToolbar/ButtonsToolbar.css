.buttons-toolbar {
    margin-bottom: var(--space-sm);
    display: flex;
    flex-grow: 1;

    &_no-margin {
        margin-bottom: 0;
    }

    button:not(:last-child) {
        margin-right: var(--space-sm);
    }
}
