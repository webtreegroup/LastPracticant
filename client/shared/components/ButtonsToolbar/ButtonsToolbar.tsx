import './ButtonsToolbar.css';

import React, { FC, memo } from 'react';
import block from 'bem-cn';
import { GridJustification } from '@material-ui/core';

interface ButtonsToolbarProps extends React.HtmlHTMLAttributes<HTMLElement> {
    justify?: GridJustification
    gutter?: number
    noMargin?: boolean
}

const b = block('buttons-toolbar');

export const ButtonsToolbar: FC<ButtonsToolbarProps> = memo(
    ({
        children,
        className,
        justify,
        noMargin,
    }) => (
        <div
            className={b({ 'no-margin': noMargin }).mix(className)}
            style={{
                justifyContent: justify,
            }}
        >
            {children}
        </div>
    ),
);
