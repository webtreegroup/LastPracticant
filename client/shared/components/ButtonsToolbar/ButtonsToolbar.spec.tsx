import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Button } from '@material-ui/core';
import { ButtonsToolbar } from './ButtonsToolbar';

afterEach(cleanup);

describe('ButtonsToolbar', () => {
    it('Render ButtonsToolbar component no children', () => {
        const { asFragment } = render(<ButtonsToolbar />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render ButtonsToolbar component with noMargin no children', () => {
        const { asFragment } = render(<ButtonsToolbar noMargin />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Render ButtonsToolbar component with noMargin with children', () => {
        const { asFragment } = render(
            <ButtonsToolbar noMargin>
                <Button variant="contained">Button 1</Button>
                <Button variant="text">Button 2</Button>
            </ButtonsToolbar>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
