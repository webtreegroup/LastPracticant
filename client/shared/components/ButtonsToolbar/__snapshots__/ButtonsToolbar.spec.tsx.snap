// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsToolbar Render ButtonsToolbar component no children 1`] = `
<DocumentFragment>
  <div
    class="buttons-toolbar"
  />
</DocumentFragment>
`;

exports[`ButtonsToolbar Render ButtonsToolbar component with noMargin no children 1`] = `
<DocumentFragment>
  <div
    class="buttons-toolbar buttons-toolbar_no-margin"
  />
</DocumentFragment>
`;

exports[`ButtonsToolbar Render ButtonsToolbar component with noMargin with children 1`] = `
<DocumentFragment>
  <div
    class="buttons-toolbar buttons-toolbar_no-margin"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Button 1
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Button 2
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
